torch/csrc/jit/codegen/cuda/test/test_gpu_fused_reduction.cpp

```diff
@@ -2148,6 +2150,309 @@ TEST_F(NVFuserTest, FusionCrossIterationGroupedGridAllreduce4_CUDA) {
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+  auto tv2 = Welford(tv1, {0}).avg;
+  auto tv3 = broadcast(tv2, {true, false});
+  auto tv4 = add(tv0, tv3);
+  fusion.addOutput(tv4);
+
+  const int vec1 = 2;
+  const int vec2 = 3;
+  const int tidx = 16;
+  const int tidy = 8;
+
+  tv1->split(1, vec1);
+  tv1->split(1, vec2);
+  tv1->split(1, tidx);
+  tv1->split(0, tidy);
+  TransformPropagator propagator(tv1);
+  MaxRootDomainInfoSpanningTree(tv1).traverse(&propagator);
+
+  tv1->axis(0)->parallelize(ParallelType::BIDy);
+  tv1->axis(1)->parallelize(ParallelType::TIDy);
+  tv1->axis(2)->parallelize(ParallelType::BIDx);
+  tv1->axis(3)->parallelize(ParallelType::TIDx);
+
+  scheduler_utils::parallelizeAllLike(tv1);
+
+  tv2->axis(4)->parallelize(ParallelType::Group);
+  tv2->axis(5)->parallelize(ParallelType::Group);
+
+  std::vector<int64_t> shape({99, 129});
+
+  // Make sure the reduction expr is converted to GroupedGridReduciton
+  // and the non-reduction domains of the output TV are either
+  // grouped or parallelized
+  GpuLower gpulw(&fusion);
+  bool validated = false;
+  for (auto expr : KernelExprVisitor::getAllExprs(gpulw.kernel())) {
+    auto grouped_grid_reduction = dynamic_cast<kir::GroupedGridWelford*>(expr);
+    if (grouped_grid_reduction == nullptr) {
+      continue;
+    }
+    validated = true;
+  }
+  TORCH_CHECK(
+      validated, "Invalid lowered kernel. No GroupedGridWelford found.");
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+  auto t0 = at::randn(shape, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {t0});
+  auto outputs = fe.runFusion({t0});
+
+  auto t0_double = t0.to(at::kDouble);
+  auto ref = t0_double + t0_double.mean({0}).unsqueeze(0);
+
+  testValidate(fe.kernel(), outputs, {t0}, {ref}, __LINE__, __FILE__);
+}
+
+// Follows the pattern of persistent outer grid welford in batchnorm
+TEST_F(NVFuserTest, FusionCrossIterationGroupedGridAllreduceWelfordShmoo_CUDA) {
+  struct Params {
+    int N;
+    int H;
+    int W;
+    int C;
+    int tidx;
+    int tidy;
+    int vect;
+    int persistent_buffer;
+    int bidx;
+  };
+
+  auto test = [](const Params& params) {
+    Fusion fusion;
+    FusionGuard fg(&fusion);
+
+    std::vector<bool> bcast_pattern{true, true, true, false};
+    std::vector<int> reduction_dims{2, 1, 0};
+
+    auto tv0 = makeSymbolicTensor(4);
+    fusion.addInput(tv0);
+
+    auto tv1 = set(tv0);
+    auto tvs = Welford(tv1, reduction_dims);
+    auto tv2 = tvs.avg;
+    auto tv3 = tvs.var_sum;
+    auto tv4 = tvs.n;
+    auto tv5 = broadcast(tv2, bcast_pattern);
+    auto tv6 = broadcast(tv3, bcast_pattern);
+    auto tv7 = broadcast(tv4, bcast_pattern);
+    auto tv8 = sub(tv1, tv5);
+    auto tv9 = add(tv8, tv6);
+    // auto tv10 = div(tv9, tv7);
+    // fusion.addOutput(tv10);
+    fusion.addOutput(tv9);
+
+    // Schedule the fusion as it will be done by the persistent
+    // scheduler
```

Comment:
Didn't check this schedule carefully, I assume once outer grid persistence is in we'll just send it to the segmenter/schedulers.